src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api/Services/DataSetService.cs

```diff
@@ -180,4 +210,199 @@ private static DataSetVersionViewModel MapDataSetVersion(DataSetVersion dataSetV
+            .Select(MapFilterMeta)
+            .OrderBy(fm => fm.Label)
+            .ToList();
+
+        var indicators = dataSetVersion.IndicatorMetas
+            .Select(MapIndicatorMeta)
+            .OrderBy(im => im.Label)
+            .ToList();
+
+        var geographicLevels = dataSetVersion.LocationMetas
+            .Select(lm => lm.Level)
+            .ToList();
+
+        var locations = dataSetVersion.LocationMetas
+            .Select(MapLocationMeta)
+            .ToList();
+
+        var timePeriods = dataSetVersion.TimePeriodMetas
+            .Select(MapTimePeriod)
+            .OrderBy(tm => tm.Code.GetEnumValue())
+            .ThenBy(tm => tm.Period)
+            .ToList();
+
+        return new DataSetMetaViewModel
+        {
+            Filters = filters,
+            Indicators = indicators,
+            GeographicLevels = geographicLevels,
+            Locations = locations,
+            TimePeriods = timePeriods,
+        };
+    }
+
+    private static FilterMetaViewModel MapFilterMeta(FilterMeta filterMeta)
+    {
+        var options = filterMeta.Options
+            .Select(fom => MapFilterOptionMeta(filterMeta, fom))
+            .OrderBy(fom => fom.Label)
+            .ToList();
+
+        return new FilterMetaViewModel
+        {
+            Id = filterMeta.PublicId,
+            Hint = filterMeta.Hint,
+            Label = filterMeta.Label,
+            Options = options,
+        };
+    }
+
+    private static FilterOptionMetaViewModel MapFilterOptionMeta(
+        FilterMeta filterMeta,
+        FilterOptionMeta filterOptionMeta)
+    {
+        var publicId = filterOptionMeta.MetaLinks.Single(ml => ml.MetaId == filterMeta.Id).PublicId;
+
+        return new FilterOptionMetaViewModel
+        {
+            Id = publicId,
+            Label = filterOptionMeta.Label,
+            IsAggregate = filterOptionMeta.IsAggregate,
+        };
+    }
+
+    private static IndicatorMetaViewModel MapIndicatorMeta(IndicatorMeta indicatorMeta)
+    {
+        return new IndicatorMetaViewModel
+        {
+            Id = indicatorMeta.PublicId,
+            Label = indicatorMeta.Label,
+            Unit = indicatorMeta.Unit,
+            DecimalPlaces = indicatorMeta.DecimalPlaces,
+        };
+    }
+
+    private static LocationLevelMetaViewModel MapLocationMeta(LocationMeta locationMeta)
+    {
+        var options = locationMeta.Options
+            .Select(lom => MapLocationOptionMeta(locationMeta, lom))
+            .OrderBy(lom => lom.Label)
+            .ToList();
+
+        return new LocationLevelMetaViewModel
+        {
+            Level = locationMeta.Level,
+            Options = options,
+        };
+    }
+
+    private static LocationOptionMetaViewModel MapLocationOptionMeta(
+        LocationMeta locationMeta,
+        LocationOptionMeta locationOptionMeta)
+    {
+        var publicId = locationOptionMeta.MetaLinks.Single(ml => ml.MetaId == locationMeta.Id).PublicId;
```

Comment:
Similar to filter options mapping - it's inefficient to have to loop through all the links to find the public ID. 

We should be able to just pass in the `LocationOptionMetaLink` directly instead and use that i.e.

```cs
private static LocationOptionMetaViewModel MapLocationOptionMeta(
    LocationMeta locationMeta,
    LocationOptionMetaLink locationOptionMetaLink)
{
    var publicId = locationOptionMetaLink.PublicId;
    
    // ...
}
```

Reply:
I have no idea why I didn't do this before.......